### DISCUSSION TITLE
looking for a C++ function in the current buffer

### DISCUSSION THREAD

**Discussion by u56237z** [LunarVim/LunarVim#4167] [Q&A] [answered]
Hi everybody,

I have a large codebase with huge C++ files containing hundreds of functions, is there a way to jump to a specific function in the current buffer just by knowing the name (like telescope for functions)?
I can't use gd/gD because there is nothing under the cursor, gr wont do either because it will search everywhere... what I am doing right now is /func/ but it return every matching text in the buffer (so any call/mention to func as well), then I can use gd on the first occurrence..

Is there any faster way to do it? on Vim I used fuzzy ctrl-p, which is like ctrl-p but for functions. I cant find anything similar on lunarvim and I dont want to look for plugins if something is already available.

Thanks

**Comment by LostInTheLogs** [comment, marked as answer]
read the docs https://github.com/nvim-telescope/telescope.nvim#neovim-lsp-pickers

`:Telescope lsp_document_symbols`

**Reply by u56237z** [reply]
exactly was I was looking for! Thanks!
